src/basic/socket-util.c

```diff
@@ -1319,6 +1320,54 @@ int flush_accept(int fd) {
         }
 }
 
+ssize_t flush_mqueue(int fd) {
+        _cleanup_free_ char *buf = NULL;
+        struct mq_attr attr;
```

Comment:
move `attr` into innermost scope please.

Reply:
Unless I am misunderstanding you I don't think that will work since attr.mq_msgsize is needed by mq_receive().

Reply:
oh, right